features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store/source/src/app/components/Apis/Listing/ApiTableRow.js

```diff
@@ -16,63 +16,223 @@
         );
+        // Get the application names of subscribed aplications
+        for (let i = 0; i < applications.list.length; i++) {
+          let applicationId = applications.list[i].applicationId;
+          let applicationName = applications.list[i].name;
+          for (var j = 0; j < subscribedApplications.length; j++) {
+            if (subscribedApplications[j].value === applicationId) {
+              subscribedApplications[j].label = applicationName;
+            }
+          }
+        }
+        this.setState({ subscribedApplications: subscribedApplications });
+      })
+      .catch(error => {
+        if (process.env.NODE_ENV !== "production") {
+          console.log(error);
+        }
+        let status = error.status;
+        if (status === 404) {
+          this.setState({ notFound: true });
+        }
+      });
+  }
+
+  render() {
+    let details_link = "/apis/" + this.props.api.id;
+    const { name, version, context, description } = this.props.api;
+    if (!this.state.active) {
+      // Controls the delete state, We set the state to inactive on delete success call
+      return null;
     }
+    const { classes } = this.props;
+    const apiRows = [
+      <TableRow
+        hover
+        onClick={this.handleClick}
+        key={"row-data-" + this.props.api.id}
+      >
+        <TableCell className={this.state.collapseOpen ? classes.root : null}>
+          <Link to={details_link}>{name}</Link>
+        </TableCell>
+        <TableCell className={this.state.collapseOpen ? classes.root : null}>
+          {version}
+        </TableCell>
+        <TableCell className={this.state.collapseOpen ? classes.root : null}>
+          {context}
+        </TableCell>
+        <TableCell className={this.state.collapseOpen ? classes.root : null}>
+          <StarRatingBar rating={this.state.rating} />
+        </TableCell>
+      </TableRow>
+    ];
+
+    // If the state of collapsible row is open, render the row with additional details of the api
+    if (this.state.collapseOpen) {
+      apiRows.push(
+        <TableRow key={"row-data-child-" + this.props.api.id}>
+          <TableCell colSpan="4">
+            <Typography variant="body2">{this.state.description}</Typography>
+            <Typography variant="body2">
+              Policies:
+              {this.state.policies.length === 0 ? (
+                <span>&lt; NOT SET FOR THIS API &gt;</span>
+              ) : null}
+              {this.state.policies.map((policy, index) => (
+                <span key={"span-policy-" + index}>
+                  {" " + policy}
+                  {this.state.policies.length !== 1 &&
+                    index !== this.state.policies.length - 1 && <span>,</span>}
+                </span>
+              ))}
+            </Typography>
+            <Typography variant="body2">
+              Subscribed Apps:
+              {this.state.subscribedApplications.length === 0 ? (
```

Comment:
IMO we could initially do object destructuring to get the `this.state.subscribedApplications` to a constant

Reply:
Used object destructuring for this.state to get subscribedApplications to a const,